pkg/admission/validator/lakom/shoot.go

```diff
@@ -0,0 +1,85 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package cache
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/gardener-extension-shoot-lakom-service/pkg/apis/lakom"
+	extensionswebhook "github.com/gardener/gardener/extensions/pkg/webhook"
+	"github.com/gardener/gardener/pkg/apis/core"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// shoot validates shoots
+type shoot struct {
+	apiReader client.Reader
+	decoder   runtime.Decoder
+}
+
+// NewShootValidator returns a new instance of a shoot validator.
+func NewShootValidator(apiReader client.Reader, decoder runtime.Decoder) extensionswebhook.Validator {
+	return &shoot{
+		apiReader: apiReader,
+		decoder:   decoder,
+	}
+}
+
+// TODO: Some func in gardener/gardener instead of this? This was pulled from registry-cache extension
+// FindExtension finds extension with the given type.
+// The first return argument is index of the extension in the list. -1 is returned if the extension is not found.
+// The second return argument is the extension itself. An empty extension is returned if the extension is not found.
+func findExtension(extensions []core.Extension, extensionType string) (int, core.Extension) {
+	for i, ext := range extensions {
+		if ext.Type == extensionType {
+			return i, ext
+		}
+	}
+
+	return -1, core.Extension{}
+}
+
+// Validate validates the given shoot object
+func (s *shoot) Validate(_ context.Context, new, _ client.Object) error {
+	allErrs := field.ErrorList{}
+
+	shoot, ok := new.(*core.Shoot)
+	if !ok {
+		return fmt.Errorf("wrong object type %T", new)
+	}
+
+	fmt.Println("Shoot spec updated")
+
+	// TODO: Move `shoot-lakom-service` to constants
+	i, lakomExt := findExtension(shoot.Spec.Extensions, "shoot-lakom-service")
+	if i == -1 {
+		return nil
+	}
+
+	providerConfigPath := field.NewPath("spec", "extensions").Index(i).Child("providerConfig")
+	if lakomExt.ProviderConfig == nil {
+		return field.Required(providerConfigPath, "providerConfig is required for the lakom extension")
+	}
+
+	lakomConfig := &lakom.LakomConfig{}
+	if err := runtime.DecodeInto(s.decoder, lakomExt.ProviderConfig.Raw, lakomConfig); err != nil {
+		return fmt.Errorf("failed to decode providerConfig: %w", err)
+	}
+
+	// TODO(rrhubenov): I don't like this. This can easily be forgotten when a new scope is added.
+	switch lakomConfig.Scope {
+	case lakom.Cluster:
+	case lakom.KubeSystem:
+	case lakom.KubeSystemManagedByGardener:
+	default:
+		// TODO(rrhubenov): This logs the error but does not print to stderr of the client. Should be improved
+		allErrs = append(allErrs, field.Invalid(providerConfigPath.Child("scope"), lakomConfig.Scope, fmt.Sprintf("Invalid scope %s. Please refer to the documentation for available scopes", lakomConfig.Scope)))
+	}
```

Comment:
Imo, this should be refactored into an array that contains the available values and the array should be next to the definition of the constants in pkg/apis/lakom/types.go